set.mm

```diff
@@ -532760,6 +532528,73 @@ dependency is expressed in our hypothesis (called implicit
       CDBJABCDKL $.
   $}
 
+  ${
+    $d t x $.  $d x ph $.
+    $( Reverting substitution yields the original expression.  (Contributed by
+       Wolf Lammen, 5-Aug-2023.) $)
+    wl-ich-lem1 $p |- ( [ t / x ] [ x / t ] ph <-> ph ) $=
+      ( wsb sbco2vv sbid bitri ) ACBDBCDACCDAACCBEACFG $.
+  $}
+
+  ${
+    $d a x $.  $d a b y $.  $d a b ph $.
+    $( Reverting two substitutions in succession.  (Contributed by Wolf Lammen,
+       5-Aug-2023.) $)
+    wl-ich-lem2 $p |- ( [ y / b ] [ x / a ] [ a / x ] [ b / y ] ph <-> ph ) $=
+      ( wsb wl-ich-lem1 sbbii bitri ) ACEFZBDFDBFZECFJECFAKJECJDBGHAECGI $.
+  $}
+
+  $( ~ spsbim applied twice.  (Contributed by Wolf Lammen, 5-Aug-2023.) $)
+  wl-ich-lem3 $p |- ( A. a A. b ( ph <-> ps ) ->
+       ( [ y / b ] [ x / a ] ph <-> [ y / b ] [ x / a ] ps ) ) $=
+    ( wb wal wsb alcom nfa1 sp sbbid sps sylbi ) ABGZFHEHPEHZFHZAECIZFDIBECIZFD
+    IGPEFJRSTFDQFKQSTGFQABECPEKPELMNMO $.
+
+  ${
+    $d a x $.  $d a b y $.  $d a b ph $.
+    $( Lemma.  (Contributed by Wolf Lammen, 5-Aug-2023.) $)
+    wl-ich-lem4 $p |- ( A. a A. b ( [ a / x ] [ b / y ] ph
+                              <-> [ b / x ] [ a / y ] ph ) ->
+        ( ph <-> [ y / b ] [ x / a ] [ b / x ] [ a / y ] ph ) ) $=
+      ( wsb wb wal wl-ich-lem2 wl-ich-lem3 syl5bbr ) AACEFBDFZDBFECFLACDFBEFZGE
+      HDHMDBFECFABCDEILMBCDEJK $.
+  $}
+
+  ${
+    $d a b x $.  $d a b y $.  $d a b ph $.
+    $( Infer one criterium for interchangeability from another.  (Contributed
+       by Wolf Lammen, 5-Aug-2023.) $)
+    wl-ich $p |- ( A. a A. b ( [ a / x ] [ b / y ] ph
+                              <-> [ b / x ] [ a / y ] ph )
+        -> ( ph <-> [ x / a ] [ y / x ] [ a / y ] ph ) ) $=
+      ( wsb wb wal wl-ich-lem4 sbco4 syl6bb ) ACEFBDFACDFZBEFZGEHDHAMDBFECFLBCF
+      DBFABCDEIABCDDEJK $.
+  $}
+
+  ${
+    $d b x $.
+    $( Lemma.  (Contributed by Wolf Lammen, 5-Aug-2023.) $)
+    wl-ich-lem5 $p |- ( A. x x = y ->
+                         ( [ b / x ] ph <-> [ a / x ] [ b / y ] ph ) ) $=
+      ( wsb weq wal nfs1v sbf nfa1 drsb1 sbbid syl5bbr ) ABEFZOBDFBCGZBHZACEFZB
+      DFOBDABEIJQORBDPBKABCELMN $.
+  $}
+
+  ${
+    $d a b x $.  $d a b y $.  $d a b ph $.
+    $( Lemma.  To be meaningful, interchangeability should have ` x ` free in
+       ` ph ` (and for symmetry reasons ` y ` , too).  This requires ` x ` and
+       ` y ` be distinct, and the universe to have more than one element.
+       (Contributed by Wolf Lammen, 5-Aug-2023.) $)
+    wl-ich-lem6 $p |- ( ( A. x x = y /\
+          A. a A. b ( [ a / x ] [ b / y ] ph <-> [ b / x ] [ a / y ] ph ) )
+          -> F/ x ph ) $=
```

Comment:
According to @benjub's remark, the other variant to define interchangeability is `( A. x x = y \/ A. a A. b ( [ a / x ] [ b / y ] ph <-> [ b / x ] [ a / y ] ph ) )` (disjunction instead of conjunction). Then, however, `F/ x ph` won't follow necessarily.

Reply:
This is a bit stricter, since it rules out cases, instead of allowing exceptional ones, as the disjunction does.